src/debug/debug-pal/unix/diagnosticsipc.cpp

```diff
@@ -0,0 +1,143 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <pal.h>
+#include <pal_assert.h>
+#include "diagnosticsipc.h"
+#include <new>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <type_traits>
+
+DiagnosticsIpc::DiagnosticsIpc(const char *const pIpcName, const uint32_t pid)
+    : _pServerAddress(new (std::nothrow) sockaddr_un)
```

Comment:
This was a workaround because I could not compile it when I added the standard C header on the `.h` file. I did not know what is the sorcery needed here, and this unblocked me.
I will look into it.

Reply:
If you are unable to resolve within 5 minutes feel free to slap a comment on it and move on. Its a minor issue not worth more time than that : )